sphinx_search/_static/js/rtd_sphinx_search.js

```diff
@@ -71,97 +75,333 @@ const createDomNode = (nodeName, attributes) => {
+    });
+
+    return section_html;
+};
+
+/**
+ * Generate and return html structure
+ * for a sphinx domain result.
+ *
+ * @param {Object} domainData object containing the result data
+ * @param {String} page_link link of the main page. It is used to construct the section link
+ */
+const get_domain_html = (domainData, page_link) => {
+    let domain_template =
+        '<a href="<%= domain_link %>"> \
+            <div class="outer_div_page_results" id="<%= domain_id %>"> \
+                <span class="search__result__subheading"> \
+                    <%= domain_subheading %> \
+                </span> \
+                <p class="search__result__content"><%= domain_content %></p> \
+            </div> \
+        </a> \
+        <br class="br-for-hits">';
+
+    let domain_link = `${page_link}#${domainData._source.anchor}`;
+    let domain_role_name = domainData._source.role_name;
+    let domain_type_display = domainData._source.type_display;
+    let domain_doc_display = domainData._source.doc_display;
+    let domain_display_name = domainData._source.display_name;
+    let domain_name = domainData._source.name;
+
+    // take values from highlighted fields (if present)
+    if (domainData.highlight !== undefined && domainData.highlight !== null) {
+        if (
+            domainData.highlight["domains.name"] !== undefined &&
+            domainData.highlight["domains.name"].length >= 1
+        ) {
+            domain_name = domainData.highlight["domains.name"][0];
+        }
+
+        if (
+            domainData.highlight["domains.display_name"] !== undefined &&
+            domainData.highlight["domains.display_name"].length >= 1
+        ) {
+            domain_display_name =
+                domainData.highlight["domains.display_name"][0];
+        }
+
+        if (
+            domainData.highlight["domains.type_display"] !== undefined &&
+            domainData.highlight["domains.type_display"].length >= 1
+        ) {
+            domain_type_display =
+                domainData.highlight["domains.type_display"][0];
+        }
+    }
+
+    // preparing domain_content
+    let domain_content = "";
+    if (
+        typeof domain_type_display === "string" &&
+        domain_type_display.length > 0
+    ) {
+        // domain_content = type_display --
+        domain_content += domain_type_display + " -- ";
+    }
+    if (typeof domain_name === "string" && domain_name.length > 0) {
+        // domain_content = type_display -- name
+        domain_content += domain_name + " ";
+    }
+    if (
+        typeof domain_doc_display === "string" &&
+        domain_doc_display.length > 0
+    ) {
+        // domain_content = type_display -- name -- in doc_display
+        domain_content += "-- in " + domain_doc_display;
+    }
+
+    let domain_subheading = "";
+    if (
+        typeof domain_display_name === "string" &&
+        domain_display_name.length >= 2 // >= 2 because many display_names have values "-"
```

Comment:
Can we fix indexing to not store this data?

Reply:
I think we can.
I will look into this more.

Reply:
@ericholscher 
This is fixed now -- https://github.com/readthedocs/readthedocs.org/pull/5829/commits/f9883021c84f2295679610639346c55e0f706680